tests/testthat/testEdgeThresh.R

```diff
@@ -0,0 +1,251 @@
+    #   (Note that the output needs to be independently verifiably correct.
+    #    Don't run an erroneous function, and then test against the erroneous
+    #    output you received when you ran your function for the first time.)
+    # Try this with important variations of parameters (reasonable, not
+    #    combinatorially exhaustive).
+    # Cover your code.
+
+    # ((Is graph.attributes(gWithThresh)$delta the correct way of retrieving
+    # delta values from graph attributes?)))
+    ### Check edgeThresh with different valid input values
+    ### For each check, length deltaVector should be Lmax - 1, numeric, and
+    ### between 0 and 1.
+    ### Check that the input graph is not changed
+    ### Test with N & Q = 1, 5, 10, Lmax 10 & 20
+    ### ((Test with N = 10, Q = 1 -> is there a limit to how many permutations
+    # there can be if only |E| edge swaps take place?))
+
+    gWithThresh <- edgeThresh(myEGG, Lmax = 20, N = 1, Q = 1)
+    deltaVector <- igraph::graph.attributes(gWithThresh)$delta
+
+    # Check that the input graph is not changed
+    # ((what is the fastest way to check this?))
+    expect_true(isomorphic(gWithThresh, myEGG))
+
+    # Check delta length, mode and range
+    expect_equal(length(deltaVector), 19)
+    expect_equal(mode(deltaVector), "numeric")
+    expect_true(all(deltaVector[] >  0))
+    expect_true(all(deltaVector[] <= 1))
+
+    ###
+    gWithThresh <- edgeThresh(myEGG, Lmax = 20, N = 5, Q = 5)
+    deltaVector <- igraph::graph.attributes(gWithThresh)$delta
+
+    expect_true(isomorphic(gWithThresh, myEGG))
+
+    expect_equal(length(deltaVector), 19)
+    expect_equal(mode(deltaVector), "numeric")
+    expect_true(all(deltaVector[] >  0))
+    expect_true(all(deltaVector[] <= 1))
+
+
+    ###
+    gWithThresh <- edgeThresh(myEGG, Lmax = 20, N = 10, Q = 10)
+    deltaVector <- igraph::graph.attributes(gWithThresh)$delta
+
+    expect_true(isomorphic(gWithThresh, myEGG))
+
+    expect_equal(length(deltaVector), 19)
+    expect_equal(mode(deltaVector), "numeric")
+    expect_true(all(deltaVector[] >  0))
+    expect_true(all(deltaVector[] <= 1))
+
+
+    ###
+    gWithThresh <- edgeThresh(myEGG, Lmax = 10, N = 10, Q = 1)
+    deltaVector <- igraph::graph.attributes(gWithThresh)$delta
+
+    expect_true(isomorphic(gWithThresh, myEGG))
+
+    expect_equal(length(deltaVector), 9)
+    expect_equal(mode(deltaVector), "numeric")
+    expect_true(all(deltaVector[] >  0))
+    expect_true(all(deltaVector[] <= 1))
+
+})
+
+
+test_that("a corrupt input to edgeThresh does not lead to corrupted output", {
+    # Try: - spurious characters in numeric columns ("N/A", ...).
+    #      - extra tabs at line-end
+    #      - comments before headers
+    #      - truncated file (incomplete last line)
+    # Again: make absolutely sure be you never have an erroneous
+    #        one-time execution of a loop
+
+    # Test with invalid EGG graphs
+    # ((How can one create an invalid EGG graph?))
```

Comment:
What assumptions are you making about the graph? An invalid graph is one that violates your assumptions.